### ISSUE TITLE
issue: Accessing form state disables default values

### ISSUE THREAD

**Issue by ingadi** [iway1/react-ts-form#88]
### Version Number

v.1.4.2

### Codesandbox/Expo snack

https://codesandbox.io/s/black-mountain-14xil1?file=/src/App.tsx

### Steps to reproduce

1. Default value 'hello' appears in text field.
2. Go to line 35 - (where you pass the result of useForm() to your form), uncomment the line and refresh.
3. Default value doesn't appear in the text field anymore.

### Expected behaviour

Default values passed into the form should appear even when form state is accessed.

**Comment by iway1** [comment]
fixed in v1.4.3, thanks for the repro

**Comment by ingadi** [comment]
Just upgraded, but now getting a warning 'Cannot update a component (...) while rendering a different component (...).

**Comment by iway1** [comment]
shit. going to revert for now

**Comment by iway1** [comment]
you can use

```
useForm({
  defaultValues: {}
})
```
for now. I'm not sure there's a way to make it work with the prop if that didn't work

**Comment by ingadi** [comment]
Before the fix my work around was skipping default values altogether and using reset with useEffect to set the form values on mount.

```
  const form = useForm<z.infer<typeof FormSchema>>();
  const { reset } = form;

  useEffect(() => reset({ text: "hello" }), [reset]);

  return (
    <div className="App">
      <Form
        form={form}
        schema={FormSchema}
        onSubmit={(data) => {
          console.log("Got the data");
          console.log(data);
        }}
        renderAfter={() => <button>Submit</button>}
      />
    </div>
  );
```

**Comment by iway1** [comment]
That's what I should've done first =D. 

Just released v1.4.5 with this included
